clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts

```diff
@@ -35,14 +37,24 @@ import { Utils } from "../util/Utils";
 import { CopybookURI } from "./CopybookURI";
 import { CopybookProfile, DownloadQueue } from "./DownloadQueue";
 import { TelemetryEventMeasurements } from "../reporter/model/TelemetryEvent";
+import {
+  EndevorElement,
+  e4eResponse,
+  profileAsString,
+} from "../../type/e4eApi.d";
+import { getProgramNameFromUri } from "../util/FSUtils";
+import { E4ECopybookService } from "./E4ECopybookService";
 
 export class CopybookName {
   public constructor(public name: string, public dialect: string) {}
 }
 
 const experimentTag = "experiment-tag";
 export class CopybookDownloadService implements vscode.Disposable {
-  public constructor(public storagePath: string) {}
+  public constructor(
+    private storagePath: string,
+    private outputChannel: vscode.OutputChannel,
```

Comment:
Every other outputChannel arguments in the PR seem to accept channel|undefined type, if it was here as well, you would not have to modify all the other unittests

Reply:
Here it must be defined while on others it can be optional because it is not being used.

Reply:
Why shouldn't it be possible to create the service without a logger?

Reply:
The service is using it. Instance of a logger can be retrieved via helper function etc.. but what would be the advantage ?

Reply:
Every other function is doing `output?.appendLine(...)` why not this one?

Reply:
These are two different services. In this service it must be defined. While on `E4ECopybookService`It isn't because the client might assembled already and does not use the log again. Missed the
`this.outputChannel?.appendLine(`Failed to find ${copybook} in Endevor`);` modified at : ba2693fb956fafd529cd684af2f2cdc7ba9276cb